cmd/soroban-cli/src/commands/tx/edit.rs

```diff
@@ -0,0 +1,145 @@
+use std::{
+    env, fs,
+    io::{stdin, Cursor, IsTerminal, Write},
+    path::PathBuf,
+    process::{self, Stdio},
+};
+
+use base64::{engine::general_purpose::STANDARD as BASE64, Engine as _};
+use stellar_xdr::curr;
+
+use crate::{commands::global, print::Print};
+
+const DEFAULT_JSON: &str = r#"{
+  "tx": {
+    "tx": {
+      "source_account": "",
+      "fee": 100,
+      "seq_num": 0,
+      "cond": "none",
+      "memo": "none",
+      "operations": [],
+      "ext": "v0"
+    },
+    "signatures": []
+  }
+}
+"#;
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error(transparent)]
+    Io(#[from] std::io::Error),
+
+    #[error(transparent)]
+    StellarXdr(#[from] stellar_xdr::curr::Error),
+
+    #[error(transparent)]
+    SerdeJson(#[from] serde_json::Error),
+
+    #[error(transparent)]
+    Base64Decode(#[from] base64::DecodeError),
+
+    #[error("Editor returned non-zero status")]
+    EditorNonZeroStatus,
+}
+
+// Command to edit the transaction
+/// e.g. `stellar tx new manage-data --data-name hello --build-only | stellar tx edit`
```

Comment:
I think `edit` should accept stdin/file/string as an input, the same way other `tx` commands do (using `tx_envelope_from_input`)

Reply:
it will, i'm just doing in parts so the pr is smaller